rsocket/RSocketServer.cpp

```diff
@@ -161,11 +162,32 @@ void RSocketServer::onRSocketResume(
   }
   auto serverState = std::move(result.value());
   CHECK(serverState);
+  auto* eventBase = folly::EventBaseManager::get()->getExistingEventBase();
+  VLOG(2) << "Resuming client on " << eventBase->getName();
   if (!serverState->eventBase_.isInEventBaseThread()) {
-    throw RSocketException("Trying to RESUME in a different worker thread");
+    // If the resumed connection is on a different EventBase, then use
+    // ScheduledFrameTransport and ScheduledFrameProcessor to ensure the
+    // RSocketStateMachine continues to live on the same EventBase and the
+    // IO happens in the new EventBase
+    auto scheduledFT = yarpl::make_ref<ScheduledFrameTransport>(
```

Comment:
Can we move the creation of scheduledFT into the lambda?

Reply:
No strong preference either way.  I found this easier to read.